R/collect.R

```diff
@@ -60,9 +84,217 @@ collect_predictions <- function(x) {
     rlang::abort(msg)
   }
 
-  collector(x, coll_col = coll_col)
+  x <- filter_predictions(x, parameters)
+
+  if (summarize) {
+    x <- average_predictions(x, parameters)
+  } else {
+    x <- collector(x, coll_col = coll_col)
+  }
+  x
+}
+
+filter_predictions <- function(x, parameters) {
+  if (is.null(parameters)) {
+    return(x)
+  }
+  metrics <- attr(x, "metrics")
```

Comment:
Is `metrics` used other than to check if it exists? Does the check on `params` cover that?